### PR TITLE
Increase reliability of server tests by upping the wait time for proc…

### PR DESCRIPTION
**Pull request by agocke**
…ess exit

### CODE REVIEW

File: src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs [hunk 1]
```diff
@@ -2411,9 +2411,8 @@ public void MutexStopsServerStarting()
                     var result = ProcessUtilities.StartProcess(_compilerServerExecutable,
                                                                $"-pipename:{pipename}");
 
-                    // Note: can't use async because Mutexes are thread-affinitive
-                    Thread.Sleep(TimeSpan.FromMilliseconds(500));
-                    var exited = result.HasExited;
+                    // Wait up to 5 seconds for the process to exit
+                    var exited = result.WaitForExit(TimeSpan.FromSeconds(5).Milliseconds);
```
Comment (jaredpar) [inline src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs]: Can we make this a static somewhere?  It's used at least twice in the code.

**Comment by agocke** [comment]
I'm just playing around with this one for now. I wrote it last night on my Mac so I haven't even compiled the code

**Comment by agocke** [comment]
This is ready for review now @jaredpar @TyOverby @gafter @AlekseyTs @VSadov 

This should fix #5572. Rather than introducing fixed delays (which I was scared to make too long and slow down our tests runs) we now simply wait a maximum amount of time.

This should result in fast tests when the machines are fast, which is the best we can ask for.

**Comment by agocke** [comment]
@dotnet/roslyn-compiler Review please

**Comment by TyOverby** [comment]
LGTM

**Comment by agocke** [comment]
@dotnet/roslyn-compiler Can I get one more review?

File: src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs [hunk 2]
```diff
@@ -2423,24 +2425,45 @@ public void MutexStopsServerStarting()
...
+                {
+                    break;
+                }
+
+                if (DateTime.Now.Subtract(start) >= s_fiveSec)
+                {
+                    Assert.True(false, "server took more than 5 seconds to start, please investigate");
+                }
+            }
+            Assert.False(server1.HasExited);
```
Comment (AlekseyTs) [inline src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs]: Why do we assume the server hasn't exited?

Comment (agocke) [inline src/Compilers/Core/VBCSCompilerTests/CompilerServerTests.cs]: Default "live" time in these tests is 30s

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**